gpflow/multikernels.py

```diff
@@ -1,25 +1,40 @@
 import tensorflow as tf
 
+from . import settings
 from .decors import params_as_tensors
 from .features import InducingPoints, InducingFeature
 from .kernels import Kernel, Combination
 from .params import Parameter
 
 
-class IndependentMultiKernel(Kernel):
+class MultiKernel(Kernel):
     pass
 
 
-class MultiKernel(Kernel):
+class IndependentMultiKernel(MultiKernel):
     pass
 
 
 class IndependentFeature(InducingFeature):
     pass
 
 
+class MultiInducingPoints(InducingPoints):
+    @params_as_tensors
+    def Kuu(self, kern, jitter=0.0):
+        Kzz = kern.K(self.Z)
+        num_inducing_variables = tf.shape(Kzz)[0] * tf.shape(Kzz)[1]
+        Kzz += jitter * tf.reshape(tf.eye(num_inducing_variables, dtype=settings.dtypes.float_type), tf.shape(Kzz))
+        return Kzz
+
+    @params_as_tensors
+    def Kuf(self, kern, Xnew):
+        Kzx = kern.K(self.Z, Xnew)
+        return Kzx
+
+
 class Independent(Combination, IndependentMultiKernel):
-    def K(self, X, X2=None, presliced=False, full_cov_output=False):
+    def K(self, X, X2=None, presliced=False, full_cov_output=True):
```

Comment:
I just thought maybe I keep finding `full_cov_output` confusing because "full output-covariance" makes more sense than "full covariance output" (which is what I keep reading into this instead of "full covariance *between the* outputs"? Before we add it all to gpflow-master maybe we can spend a bit more time thinking about how to name things so they make sense...